docs/instructor-training-workflow.txt

```diff
@@ -0,0 +1,249 @@
+  - Admin selects <Training Task>.
+  - Admin:
+    - Enters date of interactive event.
+    - Adds comments (if any).
+  - Admin indicates whether <Training Task> is now complete.
+
+* [admin reports bulk completion of interactive work]: for all participants.
+  - Admin views {training task completion}.
+  - Admin selects one or more <Person>s.
+  - Admin selects <Training Task> (e.g., discussion, demo).
+  - AMY marks all <Training Task>s as "Completed".
+
+## Completing training work.
+
+* [admin changes completion status of work for trainee]
+  - Admin looks up <Training Progress> associated with trainee.
+  - Admin changes "Completed" status of work item(s).
+  > [training automatically completed]
+
+* [admin finds work to review]
+  - Admin views {training progress summary}.
+  - Admin selects <Training Progress>.
+  > [admin changes completion status of work for trainee].
+
+## Completing training.
+
+* [training automatically completed]: triggered by changes in <Training Task>s.
+  - AMY records change to <Training Task> and all <Training Task>s for <Person> now complete.
+  - AMY displays {badge creation form}.
+  - Admin confirms creation of badge.
+  - AMY marks <Training Progress> as "Completed".
+  - AMY creates certificate.
+  - AMY mails certificate to <Person>.
+
+## Ending training.
+
+* [admin ends training]: because trainee has gone silent, failed a work item, etc.
+  - Admin views {individual training progress}.
+  - Admin selects <Training Progress>.
+  - Admin marks <Training Progress> as "Unsuccessful".
+  - AMY marks <Training Progress> as "Completed".
+
+* [admin ends training course]: because it has timed out.
+  - Admin selects training <Event>.
+  - Admin marks <Event> as "Stalled".
+  - AMY marks all associated <Training Progress>es as "Unsuccessful".
+  - AMY marks <Training Progress> as "Completed".
+
+## Questions
+
+- What to do if Admin deletes <Task> from training <Event> directly rather than via [admin removes people from training event]?
+- Do we only allow feedback loop/multiple tries for interactive work (discussions and demos), but *not* for written work?  That's how the workflows are written now...
+- Can we support group applications by providing one extra free-text field in the application called "Group Name", and asking people who want to co-apply to fill it in with the same value?  We can then sort/filter/group by that value to find people in the same group.
+- Do we use one <Training Progress> with multiple fields, or have <Training Progress> point at multiple (atomic) <Training Task>s?  This doc assumes the latter.
+- Should each <Training Task> accumulate multiple date+comment pairs (one for each attempt to pass that task), or should people have multiple <Training Task> objects of a particular type (e.g., "demo"), N-1 of which are "fail", the last of which is "pass"?
+- Should we actually delete <Training Application>s that we don't want, or mark as "discarded"?
```

Comment:
I always prefer to mark records as "discarded" or "deleted" rather than remove them entirely from db, because you can revert this operation.